File: src/examples/java/BST.java

public class BST {
    BST left, right;
    final int data;

    private BST(int data, BST left, BST right) {
        this.data = data;
        this.left = left;
        this.right = right;

        memeographname = "BST(" + data + ")";
    }

    public BST() {
        data = 0;
        left = right = null;
    }

    public BST(int data) {
        this.data = data;        
        left = new Leaf();
        right = new Leaf();

        memeographname = "BST(" + data + ")";
    }

    public BST insert(int n)
    {
        if (n < data){
            left = left.insert(n);
            return this;
        }else{
            right = right.insert(n);
            return this;
        }
    }

    protected int memeographcolor = 0xCDB79E;
    protected String memeographname = "BST()";

    @Override
    public String toString()
    {
        return "(" + left + " " + data + " " + right + ")";
    }
}

File: bridges/bst.c
#include <stdio.h>
#include <stdlib.h>

struct bst {
        int data;

        struct bst* left;
        struct bst* right;
};

struct bst* insert(int d, struct bst* root)
{
        struct bst* curr = root;
        struct bst* nn = (struct bst*)malloc(sizeof(struct bst));
        nn->left = NULL;
        nn->right = NULL;
        nn->data = d;
        if (root == NULL) {
                return nn;
        }

        for (;;) {
                if (d <= curr->data) {
                        if (curr->left != NULL) {
                                curr = curr->left;
                        } else {
                                curr->left = nn;
                                return root;
                        }
                } else {
                        if (curr->right != NULL) {
                                curr = curr->right;
                        } else {
                                curr->right = nn;
                                return root;
                        }
                }
        }

        return root;
}

int main()
{
        int i;
        struct bst* tree = NULL;
        for (i = 0; i < 15; i++) {
                tree = insert(rand(), tree);
        }

        return 0;
}
